### PR TITLE
Unify reg dcerpc/v2

### PR DESCRIPTION
**Pull request by inashivb**
Partially closes [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/3315

### CODE REVIEW

File: rust/src/dcerpc/dcerpc.rs [hunk 1]
```diff
@@ -1383,12 +1389,96 @@ pub extern "C" fn rs_dcerpc_probe_tcp(direction: u8, input: *const u8,
...
+        {
+            let _ = AppLayerRegisterParser(&parser, alproto);
+        }
+        SCLogNotice!("Rust DCERPC parser registered.");
+    } else {
+        SCLogNotice!("Protocol detector and parser disabled for DCERPC.");
+    }
 }
 
+
```
Comment (victorjulien) [inline rust/src/dcerpc/dcerpc.rs:1481]: nit: excess newline

File: rust/src/dcerpc/dcerpc.rs [hunk 2]
```diff
@@ -1383,12 +1389,96 @@ pub extern "C" fn rs_dcerpc_probe_tcp(direction: u8, input: *const u8,
...
+            return;
+        }
+        if AppLayerParserConfParserEnabled(
+            ip_proto_str.as_ptr(),
+            parser.name,
+        ) != 0
+        {
+            let _ = AppLayerRegisterParser(&parser, alproto);
+        }
+        SCLogNotice!("Rust DCERPC parser registered.");
```
Comment (victorjulien) [inline rust/src/dcerpc/dcerpc.rs:1475]: SCLogDebug

File: rust/src/dcerpc/dcerpc.rs [hunk 3]
```diff
@@ -1383,12 +1389,96 @@ pub extern "C" fn rs_dcerpc_probe_tcp(direction: u8, input: *const u8,
...
+    } else {
+        SCLogNotice!("Protocol detector and parser disabled for DCERPC.");
```
Comment (victorjulien) [inline rust/src/dcerpc/dcerpc.rs:1477]: SCLogDebug

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: rust/src/dcerpc/dcerpc.rs [hunk 4]
```diff
@@ -1383,12 +1389,96 @@ pub extern "C" fn rs_dcerpc_probe_tcp(direction: u8, input: *const u8,
         if direction & (core::STREAM_TOSERVER|core::STREAM_TOCLIENT) != dir {
             unsafe { *rdir = dir };
         }
-        return 1;
+        return unsafe { ALPROTO_DCERPC };
+    }
+    return core::ALPROTO_UNKNOWN;
```
Comment (inashivb) [inline rust/src/dcerpc/dcerpc.rs:1394]: Should be FAILED

**Comment by inashivb** [comment]
@catenacyber do you also have any comments for this version?

File: rust/src/dcerpc/dcerpc.rs [hunk 5]
```diff
@@ -1363,7 +1363,7 @@ fn probe(input: &[u8]) -> (bool, bool) {
 
 #[no_mangle]
 pub extern "C" fn rs_dcerpc_probe_tcp(direction: u8, input: *const u8,
-                                      len: u32, rdir: *mut u8) -> i32
+                                      len: u32, rdir: *mut u8) -> u16
```
Comment (catenacyber) [inline rust/src/dcerpc/dcerpc.rs:1372]: Should it be even `AppProto` instead of `u16` to describe it better ?

File: rust/src/dcerpc/dcerpc.rs [hunk 6]
```diff
@@ -108,6 +108,7 @@ pub const DCERPC_TYPE_ORPHANED: u8 = 19;
 pub const DCERPC_TYPE_RTS: u8 = 20;
 pub const DCERPC_TYPE_UNKNOWN: u8 = 99;
 
+
```
Comment (catenacyber) [inline rust/src/dcerpc/dcerpc.rs:113]: nit newline

**Review by catenacyber — COMMENTED** [review]
Looks good to me besides the few nits

File: rust/src/dcerpc/dcerpc.rs (hunk 5 above)
Comment (inashivb) [reply rust/src/dcerpc/dcerpc.rs:1372]: Yes, that sounds better.

**Comment by inashivb** [comment]
Replaced with #6131

**Pull request closed without merging**